From 51b74a8ccb4ed75329af27d04d907d84a993cb86 Mon Sep 17 00:00:00 2001
From: Olivier Joris <joris.olivier@hotmail.com>
Date: Tue, 20 Apr 2021 22:10:53 +0200
Subject: [PATCH 4/8] free page_list

---
 custom/get_processes.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/custom/get_processes.c b/custom/get_processes.c
index a87642eea..5c2e1cd36 100644
--- a/custom/get_processes.c
+++ b/custom/get_processes.c
@@ -225,6 +225,8 @@ static int task_list_add(
 static void task_list_destroy(struct task_struct_list* list){
     struct task_struct_list_node* node;
     struct task_struct_list_node* next;
+    struct page_list* currentPg;
+    struct page_list* nextPg;
 
     if(!list)
         return;
@@ -237,6 +239,14 @@ static void task_list_destroy(struct task_struct_list* list){
 
     node = list->head;
     while(node){
+        currentPg = node->pgList;
+        while(currentPg){
+            if(currentPg->page)
+                kfree(currentPg->page);
+            nextPg = currentPg->next;
+            kfree(currentPg);
+            currentPg = nextPg;
+        }
         next = node->next;
         kfree(node);
         node = next;
-- 
2.27.0

